guides/source/plugins.md

```diff
@@ -85,454 +216,575 @@ $ bin/test
+the column name, you can do the following:
 
-Add an "acts_as" Method to Active Record
-----------------------------------------
+First, generate some models in your "dummy" Rails application to test this
+functionality. Run the following commands from the `test/dummy` directory:
 
-A common pattern in plugins is to add a method called `acts_as_something` to models. In this case, you
-want to write a method called `acts_as_yaffle` that adds a `squawk` method to your Active Record models.
+```bash
+$ cd test/dummy
+$ bin/rails generate model Product last_requested_at:datetime last_api_call:datetime
+$ bin/rails db:migrate
+```
 
-To begin, set up your files so that you have:
+Now update the Product model so that it acts like an API resource:
 
 ```ruby
-# yaffle/test/acts_as_yaffle_test.rb
-
-require "test_helper"
+# test/dummy/app/models/product.rb
 
-class ActsAsYaffleTest < ActiveSupport::TestCase
+class Product < ApplicationRecord
+  acts_as_api_resource api_timestamp_field: :last_api_call
 end
 ```
 
+To make the plugin available to all models, include the module in
+`ApplicationRecord` (we'll look at doing this automatically later):
+
 ```ruby
-# yaffle/lib/yaffle.rb
+# test/dummy/app/models/application_record.rb
 
-require "yaffle/version"
-require "yaffle/railtie"
-require "yaffle/core_ext"
-require "yaffle/acts_as_yaffle"
+class ApplicationRecord < ActiveRecord::Base
+  include ApiBoost::ActsAsApiResource
 
-module Yaffle
-  # Your code goes here...
+  self.abstract_class = true
 end
 ```
 
-```ruby
-# yaffle/lib/yaffle/acts_as_yaffle.rb
+Now you can test this functionality in the Rails console:
 
-module Yaffle
-  module ActsAsYaffle
-  end
-end
+```irb
+irb> Product.api_timestamp_field
+=> "last_api_call"
 ```
 
-### Add a Class Method
+### Add an Instance Method
 
-This plugin will expect that you've added a method to your model named `last_squawk`. However, the
-plugin users might have already defined a method on their model named `last_squawk` that they use
-for something else. This plugin will allow the name to be changed by adding a class method called `yaffle_text_field`.
+This plugin adds an instance method called `track_api_request` to any Active
+Record model that calls `acts_as_api_resource`. This method sets the value of
+the configured timestamp field to the current time (or a custom time if
+provided), allowing you to track when an API request was made.
 
-To start out, write a failing test that shows the behavior you'd like:
+To add this behavior, update `acts_as_api_resource.rb`:
 
 ```ruby
-# yaffle/test/acts_as_yaffle_test.rb
+# api_boost/lib/api_boost/acts_as_api_resource.rb
 
-require "test_helper"
+module ApiBoost
+  module ActsAsApiResource
+    extend ActiveSupport::Concern
 
-class ActsAsYaffleTest < ActiveSupport::TestCase
-  def test_a_hickwalls_yaffle_text_field_should_be_last_squawk
-    assert_equal "last_squawk", Hickwall.yaffle_text_field
-  end
+    included do
```

Comment:
There is no need to use `included` here. That is actually a bad practice. If you want to add an instance method to a class that include a concern, you just define it as an instance method in the concern.